db/db_impl/db_impl_open.cc

```diff
@@ -991,9 +991,11 @@ Status DBImpl::RecoverLogFiles(const std::vector<uint64_t>& wal_numbers,
 
     TEST_SYNC_POINT_CALLBACK("DBImpl::RecoverLogFiles:BeforeReadWal",
                              /*arg=*/nullptr);
+    uint64_t record_checksum;
```

Comment:
I prefer this style too but sanitizers struggle with it. Hope it works out here.